website/addons/base/__init__.py

```diff
@@ -595,7 +595,7 @@ def get_nodes_with_oauth_grants(self, external_account):
         return (
             Node.load(node_id)
             for node_id, grants in self.oauth_grants.iteritems()
-            if external_account._id in grants.keys()
+            if (external_account._id in grants.keys() and not Node.load(node_id).is_deleted)
```

Comment:
We already load the node on ~L596, so it's duplicative to load it here, too. So you might want to refactor this into a `for` loop, instead.